src/Playwright/Core/Request.cs

```diff
@@ -130,5 +122,38 @@ internal Request(IChannelOwner parent, string guid, RequestInitializer initializ
 
             return JsonDocument.Parse(content).RootElement;
         }
+
+        public async Task<RequestSizesResult> SizesAsync()
+        {
+            if (await ResponseAsync().ConfigureAwait(false) is not IChannelOwner<Response> res)
+            {
+                throw new PlaywrightException("Unable to fetch sizes for a failed request.");
+            }
+
+            return await ((ResponseChannel)res.Channel).SizesAsync().ConfigureAwait(false);
+        }
+
+        public async Task<Dictionary<string, string>> AllHeadersAsync()
+            => (await GetRawHeadersAsync().ConfigureAwait(false)).Headers;
+
+        public async Task<IReadOnlyList<RequestHeadersArrayResult>> HeadersArrayAsync()
+            => (await GetRawHeadersAsync().ConfigureAwait(false)).HeadersArray.ConvertAll(x => new RequestHeadersArrayResult() { Name = x.Name, Value = x.Value });
+
+        public async Task<string> HeaderValueAsync(string name)
+            => (await GetRawHeadersAsync().ConfigureAwait(false)).Get(name);
+
+        private async Task<RawHeaders> GetRawHeadersAsync()
+        {
+            if (_rawHeaders != null) return _rawHeaders;
```

Comment:
You need to cache the Task, not the value. Otherwise you are sending request multiple times.